projects/optic/src/commands/api/add.ts

```diff
@@ -0,0 +1,215 @@
+import { Command } from 'commander';
+import { wrapActionHandlerWithSentry } from '@useoptic/openapi-utilities/build/utilities/sentry';
+
+import { OpticCliConfig, VCS } from '../../config';
+import { getFileFromFsOrGit, ParseResult } from '../../utils/spec-loaders';
+import { logger } from '../../logger';
+import { OPTIC_URL_KEY } from '../../constants';
+
+const exec = promisify(callbackExec);
+
+const usage = () => `
+  optic api add
+  optic api add <path_to_spec.yml>
+  optic api add <path_to_spec.yml> --history-depth 0
+  optic api add <path_to_spec.yml> --ruleset <ruleset_id> --web`;
+
+const helpText = `
+Example usage:
+  Add a single api to optic
+  $ optic api add <path_to_spec.yml>
+
+  Add a single api to optic and crawl through the history \`depth\` steps. history-depth=0 will crawl the entire history
+  $ optic api add <path_to_spec.yml> --history-depth <depth>
+
+  Discover all apis in the current repo
+  $ optic api add
+
+  Add all apis and attach rulesets - configure your ruleset in Optic cloud
+  $ optic api add --ruleset <ruleset_id>
+  `;
+
+export const registerApiAdd = (cli: Command, config: OpticCliConfig) => {
+  cli
+    .command('add')
+    .configureHelp({
+      commandUsage: usage,
+    })
+    .addHelpText('after', helpText)
+    .description('Add APIs to Optic')
+    .argument('[spec_path]', 'path to file to compare')
+    .option(
+      '--history-depth <history-depth>',
+      'Set rulesets to run on API diffs. You can always add this later by setting the `[x-optic-ruleset]` key on your OpenAPI spec',
+      '1'
+    )
+    .option(
+      '--ruleset <ruleset>',
+      'Set rulesets to run on API diffs. You can always add this later by setting the `[x-optic-ruleset]` key on your OpenAPI spec'
+    )
+    .option('--web', 'open to the added API in Optic Cloud', false)
+    .action(wrapActionHandlerWithSentry(getApiAddAction(config)));
+};
+
+type ApiAddActionOptions = {
+  historyDepth: string;
+  ruleset?: boolean;
+  web?: boolean;
+};
+
+const getApiAddAction =
+  (config: OpticCliConfig) =>
+  async (path_to_spec: string | undefined, options: ApiAddActionOptions) => {
+    if (isNaN(Number(options.historyDepth))) {
+      logger.error(
+        '--history-depth is not a number. history-depth must be a number'
+      );
+      process.exitCode = 1;
+      return;
+    } else if (!path_to_spec && options.historyDepth !== '1') {
+      logger.error(
+        'Invalid argument combination: Cannot set a history-depth !== 1 when no spec path is provided'
+      );
+      process.exitCode = 1;
+      return;
+    } else if (!config.isAuthenticated) {
+      logger.error('Must be logged in to add APIs. Log in with `optic login`');
+      process.exitCode = 1;
+      return;
+    } else if (!config.vcs || config.vcs?.type !== VCS.Git) {
+      logger.error('Must have git in path and be in a git repo to add apis');
+      process.exitCode = 1;
+      return;
+    }
+
+    if ('todo fetch orgs related to token') {
```

Comment:
@nalanj I think we'll need an endpoint that returns orgs a token has access to here